providers/universal-provider/src/providers/cardano.ts

```diff
@@ -77,13 +77,15 @@ class CardanoProvider implements IProvider {
       return [];
     }
 
-    return (
-      accounts
-        // get the accounts from the active chain
-        .filter((account) => account.split(":")[1] === this.chainId.toString())
-        // remove namespace & chainId from the string
-        .map((account) => account.split(":")[2]) || []
-    );
+    return [
+      ...new Set(
+        accounts
+          // get the accounts from the active chain
+          .filter((account) => account.split(":")[1] === this.chainId.toString())
+          // remove namespace & chainId from the string
+          .map((account) => account.split(":")[2]) || [],
```

Comment:
Realise you're just wrapping with `Set` here, but if `accounts` is always an array the `|| []` is redundant, since functional operations like filter/map will always yield at least an empty array back